event-schema/event_schema.go

```diff
@@ -168,7 +181,11 @@ func (manager *EventSchemaManagerT) updateEventModelCache(eventModel *EventModel
 	if !ok {
 		manager.eventModelMap[writeKey][eventType] = make(map[string]*EventModelT)
 	}
-	manager.eventModelMap[writeKey][eventType][eventIdentifier] = eventModel
+	if len(manager.eventModelMap[writeKey]) < eventModelLimit {
+		manager.eventModelMap[writeKey][eventType][eventIdentifier] = eventModel
+	} else {
+		stats.NewTaggedStat("eventSchemas.droppedEventModelsCount", stats.CountType, stats.Tags{}).Increment()
```

Comment:
As per convention, we use snake case for stat names.

Reply:
Let's return if we are not handling it?

Reply:
Done